rules/semantic-versioning/rule.md

```diff
@@ -0,0 +1,61 @@
+---
+type: rule
+title: Do understand the value of Semantic Versioning?
+uri: semantic-versioning
+authors:
+  - title: Jason Taylor
+    url: https://www.ssw.com.au/people/jason-taylor/
+  - title: Piers Sinclair
+    url: https://www.ssw.com.au/people/piers-sinclair
+related:
+  - packages-up-to-date
+  - only-export-what-is-necessary
+created: 2022-02-02T05:56:54.309Z
+guid: 85ee49a3-ac86-432b-8608-1056b42db8bf
+---
+Semantic versioning (sometimes called SemVer) allows library developers to communicate with those who use the library. In the old days, people would choose version numbers on gut feel or auto generated. With Semantic Versioning, the version number conveys the type of changes since the last release and if any changes break backwards compatibility. That way any developer who looks at it knows the risk level when they update...
+            
+<!--endintro-->
+
+The structure of Semantic Versioning includes
+* A major version e.g. 1.0.0 or 2.0.0
+* A minor version e.g. 1.1.0 or 1.2.0
+* A patch version e.g. 1.1.1 or 1.1.2
+
+Every release should have a new version so that when users provide feedback or bug reports you know the version they were using.
+
+## How it works - When should I increment each version?
+
+### Patch version
+If the current version is 1.1.0 and the new release includes only backwards-compatible bug fixes then the new version number will be 1.1.1 (the patch version has been incremented by 1)
+
+### Minor version
+
+Following on from this, if a new release includes a new backwards-compatible feature the minor version would be incremented by 1 and the patch version is reset to 0, 1.2.0
+
+### Major version
+
+The major version will only be incremented when releasing non-backwards compatible changes. For example, when something has become obsolete or an interface has changed fundamentally.
```

Comment:
Should mention that minor and patch version reset to 0 when this is incremented.